sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -158,7 +158,7 @@ def _get_list_element_type(typ: Optional[type]) -> Optional[type]:
         if len(args) == 1:
             return args[0]
 
-    raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
+    log.error(f"Unexpected type. Expected 'list' got '{typ}'")
```

Comment:
This was the most obvious change to make to get around this issue. I'm very open to a more general fix which allows us to maintain strict type checking here, but which does not cause the affected resource to fail.